apps-rendering/src/components/TableOfContents/index.tsx

```diff
@@ -0,0 +1,188 @@
+import type { SerializedStyles } from '@emotion/react';
+import { css } from '@emotion/react';
+import { text } from '@guardian/common-rendering/src/editorialPalette/text';
+import type { ArticleFormat } from '@guardian/libs';
+import {
+	line,
+	neutral,
+	remSpace,
+	textSans,
+} from '@guardian/source-foundations';
+import {
+	SvgChevronDownSingle,
+	SvgChevronUpSingle,
+} from '@guardian/source-react-components';
+import Anchor from 'components/Anchor';
+import ListItem from 'components/ListItem';
+import OrderedList from 'components/OrderedList';
+import type { Outline } from 'outline';
+import type { FC, ReactElement } from 'react';
+import { darkModeCss } from 'styles';
+
+interface Props {
+	format: ArticleFormat;
+	outline: Outline;
+}
+
+interface TextElementProps {
+	node: Node;
+	key: number;
+}
+
+const anchorStyles = (format: ArticleFormat): SerializedStyles => css`
+	color: ${text.paragraph(format)};
+	border-bottom: none;
+	:hover {
+		border-bottom: 0.0625rem solid ${neutral[86]};
+	}
```

Comment:
Perhaps I'm missing a use-case here, but when do we expect to see the `:hover` pseudoclass activated from mobile devices?

Reply:
Mobile devices can have pointers too! 🙂

https://developer.apple.com/design/human-interface-guidelines/inputs/pointing-devices/
https://support.apple.com/en-us/HT210546

Reply:
D'oh, of course! I think I wrote this in response to Firefox's responsive viewer supressing the `:hover` states, but it makes sense to keep this selector!